gammapy/modeling/models/spectral.py

```diff
@@ -316,11 +319,29 @@ def plot(
 
         energy_min, energy_max = energy_range
         energy = MapAxis.from_energy_bounds(
-            energy_min, energy_max, n_points, energy_unit
-        ).edges
+                                            energy_min, energy_max, n_points, energy_unit
+                                            ).edges
+
+        if sed_type:
+            if sed_type == "dnde":
+                flux = self(energy).to(flux_unit)
+
+            elif sed_type == "e2dnde":
+                energy_power = 2
```

Comment:
I think the `energy_power` should remain independent, so just remove it here and handle the case "manually" like `energy ** 2 * self(energy).to(flux_unit)`

Reply:
Ok.